### PR TITLE
Add port 8080 to CNP

### PR DESCRIPTION
**Pull request by vxav**

### CODE REVIEW

**Comment by QuentinBisson** [comment]
That looks alright but I'm not even sure what the port 3101 is :D

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuentinBisson** [comment]
@marieroque do you remember ?

**Review by QuantumEnigmaa — APPROVED** [review]

**Comment by QuentinBisson** [comment]
Apparently not:

```
k describe pod -n kube-system grafana-agent-2cc4q
Name:             grafana-agent-2cc4q
Namespace:        kube-system
Priority:         0
Service Account:  grafana-agent
Node:             ip-10-224-180-202.eu-north-1.compute.internal/10.224.180.202
Start Time:       Mon, 15 Jan 2024 10:27:52 +0100
Labels:           app.kubernetes.io/instance=grafana-agent
                  app.kubernetes.io/name=grafana-agent
                  controller-revision-hash=6dbd94fbdb
                  pod-template-generation=2
Annotations:      kubernetes.io/psp: restricted
Status:           Running
IP:               100.64.11.70
IPs:
  IP:           100.64.11.70
Controlled By:  DaemonSet/grafana-agent
Containers:
  grafana-agent:
    Container ID:  containerd://959b1e32cfc95705f1c49cb0426cfe7a8f0248207b3a3cb0352f253a11706615
    Image:         gsoci.azurecr.io/giantswarm/grafana-agent:v0.37.2
    Image ID:      docker.io/giantswarm/grafana-agent@sha256:bf35f8635344ae2878eb9070b0c12bd62eb7d52f136d6aafe91c9635f0779d53
    Port:          8080/TCP
    Host Port:     0/TCP
    Args:
      run
      /etc/agent/config.river
      --storage.path=/tmp/agent
      --server.http.listen-addr=0.0.0.0:8080
    State:          Running
      Started:      Mon, 15 Jan 2024 10:27:53 +0100
    Ready:          True
    Restart Count:  0
    Readiness:      http-get http://:8080/-/ready delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      AGENT_MODE:  flow
      HOSTNAME:     (v1:spec.nodeName)
    Mounts:
      /etc/agent from config (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-9rf4x (ro)
  config-reloader:
    Container ID:  containerd://6a878d361802b2a421d48e414d90b919915e89b29973edfe200d7345a8dcb098
    Image:         gsoci.azurecr.io/giantswarm/configmap-reload:v0.8.0
    Image ID:      docker.io/giantswarm/configmap-reload@sha256:084de2d3533f9215eceef9a1feccfc11cad43cf382ea82ddfa4272f68df0614f
    Port:          <none>
    Host Port:     <none>
    Args:
      --volume-dir=/etc/agent
      --webhook-url=http://localhost:8080/-/reload
    State:          Running
      Started:      Mon, 15 Jan 2024 10:27:54 +0100
    Ready:          True
    Restart Count:  0
    Requests:
      cpu:        1m
      memory:     5Mi
    Environment:  <none>
    Mounts:
      /etc/agent from config (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-9rf4x (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      grafana-agent
    Optional:  false
  kube-api-access-9rf4x:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason           Age   From          Message
  ----     ------           ----  ----          -------
  Warning  PolicyViolation  30m   kyverno-scan  policy restrict-image-registries/validate-registries fail: validation error: Images must come from a trusted Giant Swarm registry. rule validate-registries failed at path /spec/containers/0/image/

```

**Comment by QuentinBisson** [comment]
I suspect this was a copy paste of this https://github.com/giantswarm/promtail-app/blob/master/helm/promtail/templates/cilium-networkpolicy.yaml

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add port 8080 to CiliumNetworkPolicy for prometheus.
```
Comment (QuentinBisson) [inline CHANGELOG.md]:
```suggestion
- Fix CiliumNetworkPolicy port from 3101 to 8080 so prometheus can scrape the grafana-agent.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
```
Comment (QuentinBisson) [inline CHANGELOG.md]:
```suggestion
### Fixed
```

File: helm/grafana-agent/templates/cilium-networkpolicy.yaml [hunk 3]
```diff
@@ -15,6 +15,8 @@ spec:
       - ports:
         - port: "3101"
```
Comment (QuentinBisson) [inline helm/grafana-agent/templates/cilium-networkpolicy.yaml]:
```suggestion
```

File: helm/grafana-agent/templates/cilium-networkpolicy.yaml [hunk 4]
```diff
@@ -15,6 +15,8 @@ spec:
...
           protocol: TCP
```
Comment (QuentinBisson) [inline helm/grafana-agent/templates/cilium-networkpolicy.yaml]:
```suggestion
```

**Comment by vxav** [comment]
Seems to be promtail metrics

```
> kgsa|grep 3101
kube-system   promtail-metrics                                       ClusterIP   None             <none>        3101/TCP                       117m
```

**Review by QuentinBisson — APPROVED** [review]

**Comment by marieroque** [comment]
> @marieroque do you remember ?

No idea about 3101.
My last change have set listen port to 8080 rather than 80 by default (which causing issue).

**Pull request merged**